pkg/transaction/sender_matcher.go

```diff
@@ -29,24 +31,39 @@ func NewMatcher(backend Backend, signer types.Signer) *Matcher {
 	}
 }
 
-func (m Matcher) Matches(ctx context.Context, tx []byte, networkID uint64, senderOverlay swarm.Address) (bool, error) {
+func (m *Matcher) Matches(ctx context.Context, tx []byte, networkID uint64, senderOverlay swarm.Address) (bool, []byte, error) {
+
 	incomingTx := common.BytesToHash(tx)
 
 	nTx, isPending, err := m.backend.TransactionByHash(ctx, incomingTx)
 	if err != nil {
-		return false, fmt.Errorf("%v: %w", err, ErrTransactionNotFound)
+		return false, nil, fmt.Errorf("%v: %w", err, ErrTransactionNotFound)
 	}
 
 	if isPending {
-		return false, ErrTransactionPending
+		return false, nil, ErrTransactionPending
 	}
 
 	sender, err := types.Sender(m.signer, nTx)
 	if err != nil {
-		return false, fmt.Errorf("%v: %w", err, ErrTransactionSenderInvalid)
+		return false, nil, fmt.Errorf("%v: %w", err, ErrTransactionSenderInvalid)
+	}
+
+	receipt, err := m.backend.TransactionReceipt(ctx, incomingTx)
+	if err != nil {
+		return false, nil, err
+	}
+
+	nextBlock, err := m.backend.HeaderByNumber(ctx, big.NewInt(0).Add(receipt.BlockNumber, big.NewInt(1)))
+	if err != nil {
+		return false, nil, err
+	}
+
+	if !bytes.Equal(receipt.BlockHash.Bytes(), nextBlock.ParentHash.Bytes()) {
+		return false, nil, nil
 	}
```

Comment:
What does this test for?

Reply:
The block hash of the block that contains the chequebook must be the parent of the block whose hash was used for the overlay

Reply:
How I see it:
You have a receipt (ln 52), calculate the nextBlock from that (57), and then compare the nextBlock parent's hash against the hash of the block for the receipt. Unless there is an error in libraries used by the backend, this should always be the case, as the root object (txHash) is used all along the way.